evalml/pipelines/time_series_pipeline_base.py

```diff
@@ -37,12 +37,12 @@ def __init__(
                 "Please specify them as a dictionary with the key 'pipeline'."
             )
         pipeline_params = parameters["pipeline"]
-        date_index = pipeline_params["date_index"]
-        if date_index is None:
-            raise ValueError("Parameter date_index cannot be None!")
         self.gap = pipeline_params["gap"]
         self.max_delay = pipeline_params["max_delay"]
         self.forecast_horizon = pipeline_params["forecast_horizon"]
+        self.date_index = pipeline_params["date_index"]
+        if self.date_index is None:
+            raise ValueError("Parameter date_index cannot be None!")
```

Comment:
Any particular reason for this change?

Reply:
Since `date_index` is a required parameter in time series configurations, I figured it would be good to have it as an instance variable alongside `gap`, `max_delay`, and `forecast_horizon`. And the position change just because I felt it's cleaner to have the instance variables together